src/SayMore/UI/ComponentEditors/SessionBasicEditor.cs

```diff
@@ -561,25 +571,34 @@ private void SetAccessCodeListAndValue()
 		private void HandleBinderTranslateBoundValueBeingRetrieved(object sender,
 			TranslateBoundValueBeingRetrievedArgs args)
 		{
-			if (!(args.BoundControl is MultiValueDropDownBox)) return;
-
-			if (args.ValueFromFile.Contains(","))
-				args.TranslatedValue = args.ValueFromFile.Replace(",", FieldInstance.kDefaultMultiValueDelimiter.ToString(CultureInfo.InvariantCulture));
-
-			if (args.BoundControl == _participants)
+			if (args.BoundControl is MultiValueDropDownBox)
 			{
-				var val = args.TranslatedValue ?? args.ValueFromFile;
-				if (!string.IsNullOrEmpty(val))
-				{
-					var participantNames = FieldInstance.GetMultipleValuesFromText(val).ToArray();
-					for (int index = 0; index < participantNames.Length; index++)
-						participantNames[index] = _personInformant.GetUiIdByName(participantNames[index]);
-
-					args.TranslatedValue = FieldInstance.GetTextFromMultipleValues(participantNames);
-				}
+				if (args.ValueFromFile.Contains(","))
+					args.TranslatedValue = args.ValueFromFile.Replace(",",
+						FieldInstance.kDefaultMultiValueDelimiter.ToString(CultureInfo.InvariantCulture));
+			} else if (args.BoundControl == _participants)
+			{
+				args.TranslatedValue = GetParticipantsWithRolesFromContributions();
 			}
 		}
 
+		private string GetParticipantsWithRolesFromContributions()
+		{
+			// We want to display the partipants with their roles, which means using data really from
+			// the contributions field, because we don't want the value really stored in participants,
+			// a list of names we maintain for backwards compatibility, to include the roles.
+			// Note that this value should only be displayed in the _participants control,
+			// not stored in the <participants> field (as was done briefly, breaking backwards
+			// compatibility and causing problems for other programs that use the file).
+			// (If the user could edit the field, we'd have to watch out that the binding helper didn't
+			// copy the edited names-with-roles into the file. But attempts to edit this field
+			// trigger a switch to the Contributors tab.)
+			var contributions = _file.GetValue(SessionFileType.kContributionsFieldName, null) as ContributionCollection;
+			var participantsWithRoles = string.Join("; ",
```

Comment:
I'm getting an exception here:
Msg: Value cannot be null.
Parameter name: source
Class: System.ArgumentNullException
Source: System.Core
Assembly: System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
Stack:    at System.Linq.Enumerable.Select[TSource,TResult](IEnumerable`1 source, Func`2 selector)
   at SayMore.UI.ComponentEditors.SessionBasicEditor.GetParticipantsWithRolesFromContributions() in C:\Projects\saymoreJT\src\SayMore\UI\ComponentEditors\SessionBasicEditor.cs:line 597
   at SayMore.UI.ComponentEditors.SessionBasicEditor.HandleBinderTranslateBoundValueBeingRetrieved(Object sender, TranslateBoundValueBeingRetrievedArgs args) in C:\Projects\saymoreJT\src\SayMore\UI\ComponentEditors\SessionBasicEditor.cs:line 581
   at SayMore.UI.ComponentEditors.BindingHelper.UpdateControlValueFromField(Control ctrl) in C:\Projects\saymoreJT\src\SayMore\UI\ComponentEditors\BindingHelper.cs:line 292
Thread: UI
Thread UI culture: en-US
Exception: System.ArgumentNullException

Reply:
If contributions is null, I think you just want to return an empty string.